src/pkg/immutabletag/dao/immutable.go

```diff
@@ -30,9 +31,30 @@ type immutableRuleDao struct{}
 func (i *immutableRuleDao) CreateImmutableRule(ir *model.ImmutableRule) (int64, error) {
 	ir.Disabled = false
 	o := dao.GetOrmer()
+	exist, err := i.checkExist(ir)
```

Comment:
This cannot avoid the duplicated records when running in parallel. How about adding the `unique` constraint on the database table, and try to catch the dup error when inserting the row?

Reply:
+1 to use `unique tag_filter`